### ISSUE TITLE
Milestone 1993792: confirmation fails when using branch develop/33801cb

### ISSUE THREAD

**Issue by c-mnd** [iotaledger/hornet#689]
**Describe the bug**
PANIC Tangle confirmMilestone: whiteflag.ComputeConfirmation failed with Error: the milestone approved an invalid bundle: bundle MDOXFKXNNBIFHILRZLZUQISYN9ZLVZCQHCQQCHHTJUHPRBWMEDRYEHLFZQRLLIWESFDDPUDPHBLZLUMCC is invalid

**To Reproduce**
Go version go1.15.2 linux/amd64.
Commit 33801cb.
I used https://dbfiles.iota.org/mainnet/hornet/2020-10-11T00%3A00%3A01Z-export.bin for starting the sync, but any correct snapshot just before MS 1993791 should do.
Run HORNET, it syncs to 1993791, then crashes.
I reproduced it four times.

**Expected behavior**
Milestone 1993792 should confirm.

**Environment information:**
 - HORNET commit 33801cb
 - OS: Ubuntu 20.04.1 LTS
 - RAM: 4 GB
 - Cores: 2 Cores
 - Type: VPS

**Additional context**
Apart from times, it looks the same every time and it's always bundle MDOXFKXNN...
```
2020-10-18T20:38:53+02:00	INFO	Tangle	New solid milestone: 1993791, 11.78 CTPS
2020-10-18T20:38:53+02:00	INFO	Tangle	Run solidity check for Milestone (1993792)...
2020-10-18T20:38:53+02:00	INFO	Tangle	Solidifier finished: txs: 86, collect: 0s, solidity 6ms, propagation: 0s, total: 7ms
2020-10-18T20:38:53+02:00	PANIC	Tangle	confirmMilestone: whiteflag.ComputeConfirmation failed with Error: the milestone approved an invalid bundle: bundle MDOXFKXNNBIFHILRZLZUQISYN9ZLVZCQHCQQCHHTJUHPRBWMEDRYEHLFZQRLLIWESFDDPUDPHBLZLUMCC is invalid
github.com/gohornet/hornet/plugins/tangle.solidifyMilestone
	/root/hornet/plugins/tangle/solidifier.go:434
github.com/gohornet/hornet/plugins/tangle.configureTangleProcessor.func3
	/root/hornet/plugins/tangle/tangle_processor.go:46
github.com/iotaledger/hive.go/workerpool.(*WorkerPool).startWorkers.func1
	/root/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200917222020-d5f10a0e336e/workerpool/workerpool.go:158
```

**Comment by muXxer** [comment]
Thx for the bugreport.

**Comment by c-mnd** [comment]
Thank you, inbox boTt 😀 .

**Comment by muXxer** [comment]
no bot :(

**Comment by c-mnd** [comment]
oops, I forgot you're always fast man!

**Comment by c-mnd** [comment]
I guess, I found the reason in 'iota.go':
kerl.Absorb should fail when the 243rd trit is non-zero https://github.com/iotaledger/iota.go/pull/178
introduces a breaking change.

**Comment by muXxer** [comment]
Thx for having a deeper look. I'm already aware of this. :)
Pls run the stable version for now.

**Comment by c-mnd** [comment]
I commented it away in kerl.Absorb() and continue from there. :D

**Comment by Wollac** [comment]
Yes, that PR (unintentionally) changes the validation criteria of the W-OTS signatures scheme by rejecting non-canonically encoded signatures. All signatures produced by IF libraries should always be canonical, but apparently "manually" created signatures exist.

IMHO it would be much nicer and cleaner to have this strict validation criterion, but when violating transactions are already confirmed this is not an option. I guess, I would propose to leave https://github.com/iotaledger/iota.go/pull/178 in place, since it makes a lot of sense to have Kerl fail on invalid input. Instead, set each 243rd trit of the signature to `0` before validation. Then, signatures technically are malleable, which is not great but for IOTA this is equivalent to re-attachments anyway.

**Comment by c-mnd** [comment]
Indeed, strict validation would be a lot better, but it needed some kind of deprecation announcement.
So https://github.com/iotaledger/iota.go/pull/204 is fine.

**Comment by Wollac** [comment]
https://github.com/iotaledger/iota.go/pull/204 has been merged, so an update of iota.go will fix this issue.

**Comment by muXxer** [comment]
Should be fixed with #690
